File: Core Java/student-app/src/com/techlabs/student/StudentConsole.java
package com.techlabs.student;

import java.util.ArrayList;
import java.util.Scanner;

import com.techlabs.studentstore.IStudentStore;

public class StudentConsole {
	private IStudentStore studentobj;

	private static Scanner input;
	private static final int AddChoice = 1;
	private static final int DisplayChoice = 2;
	private static final int SearchStudentChoice = 3;
	private static final int DeleteStudentChoice = 4;
	private static final int ExportToHTML = 5;

	public StudentConsole(IStudentStore obj) {
		studentobj = obj;
	}

	public void start() {

		input = new Scanner(System.in);
		System.out.println("Press 1 to add Student");
		System.out.println("Press 2 to Display Students");
		System.out.println("Press 3 to Search Student By Name");
		System.out.println("Press 4 to Delete Student");
		System.out.println("Press 5 to Export to HTML File");

		int choice = Integer.parseInt(input.nextLine());

		switch (choice) {
		case AddChoice:
			getDetails();
			break;
		case DisplayChoice:
			print(studentobj.get());
			break;
		case SearchStudentChoice:
			getSearch();
			break;
		case DeleteStudentChoice:
			getDelete();
			studentobj.delete(0);
			break;
		case ExportToHTML:
			studentobj.export();
			System.out.println("Data has been exported to CSV file");
			break;
		}

	}

		public void getDetails() {

		do {
			System.out.println("Enter Student FirstName :");
			String firstName = input.nextLine();

			System.out.println("Enter Student LastName :");
			String lastName = input.nextLine();

			System.out.println("Enter Student Address :");
			String address = input.nextLine();

			Student student = new Student(firstName, lastName, address);
			((IStudentStore) studentobj).add(student);

			System.out.println("Thank You !! Your details are being saved.");
			System.out.println("Press 1 to add more Student");
			System.out.println("Any Other number to exit");

		} while (Integer.parseInt(input.nextLine()) == AddChoice);
		start();
	}

	public void print(ArrayList<Student> students) {
		for (Student student : students) {
			System.out.println("Student Id:" + student.getId());
			System.out.println("Student Name:" + student.getFirstName() + " "
					+ student.getLastName());
			System.out.println("Student Location:" + student.getAddress()
					+ "\n");
		}
	}

	public void getSearch() {
		System.out.println("Enter Student Name :");
		Student result = studentobj.search(input.nextLine());

		System.out.println("Student ID:" + result.getId());
		System.out.println("Student Name:" + result.getFirstName()
				+ result.getLastName());
		System.out.println("Student Location:" + result.getAddress());
	}

	private void getDelete() {
		System.out.println("Enter Student Id to delete:");
		System.out.println(studentobj.delete(Long.parseLong(input.nextLine())));
	}
}

File: DOTNET/C#/StudentApp/StudentApp/StudentConsole.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StudentApp
{
    class StudentConsole
    {
        private const int ADD_CHOICE = 1;
        private const int DISPLAY_CHOICE = 2;
        private const int SEARCH_STUDENT_DETAILS = 3;
        private const int DELETE_STUDENT_DETAILS = 4;


        private List<Student> listOfStudents;
        private StudentService studentService = new StudentService();

        public void Start()
        {
            Console.WriteLine("Press 1 To Add Students");
            Console.WriteLine("Press 2 To Display Students");
            Console.WriteLine("Press 3 To Search Student detail");
            Console.WriteLine("Press 4 To Delete Student");

            int choice = int.Parse(Console.ReadLine());

            switch (choice)
            {
                case ADD_CHOICE :
                    GetDetails();
                    break;

                case DISPLAY_CHOICE :
                    PrintData();
                    break;

                case SEARCH_STUDENT_DETAILS:
                    GetSearch();
                    break;

                case DELETE_STUDENT_DETAILS:
                    GetDelete();
                    break;
            }
        }
        public void GetDetails()
        {
            Console.Write("Enter Student Name : ");
            String name = Console.ReadLine();

            Console.Write("Enter Student Age: ");
            int age = int.Parse(Console.ReadLine());

            Console.Write("Enter Student Location : ");
            String address = Console.ReadLine();

            var student = new Student(name, age, address);
            studentService.AddStudents(student);

            Console.WriteLine("Your Data Saved successfully....");
            Console.WriteLine();
            Start();
        }

        public void PrintData()
        {
            listOfStudents = studentService.GetStudents;
            foreach (Student stu in listOfStudents)
            {
                Console.WriteLine("Student Name: " + stu.StudentName);
                Console.WriteLine("Student Age: " + stu.Age);
                Console.WriteLine("Student Location: " + stu.Address);
                Console.WriteLine();
            }
        }

        public void GetSearch()
        {
            Console.WriteLine("Enter Student Name: ");
            String studentName = Console.ReadLine();
            Student searchedStudent = studentService.Search(studentName);

            Console.WriteLine("Student Id: "+searchedStudent.Id);
            Console.WriteLine("Student Name: "+searchedStudent.StudentName);
            Console.WriteLine("Student Age: "+searchedStudent.Age);
            Console.WriteLine("Student Location: "+searchedStudent.Address);
        }

        public void GetDelete() {
            Console.WriteLine("Enter student id to delete: ");
            Console.WriteLine(studentService.DeleteStudentData(Console.ReadLine()));
        }

    }
}
